### PR TITLE
Add support for map_key_exists

### PR DESCRIPTION
**Pull request by jainavi17**
UDF to return whether a key exists in the given map

Test plan

Added unit tests.
Build successfully using the following terminal command
- `./mvnw clean install -Dtest=TestMapKeyExists -fn -pl presto-main`

```
== RELEASE NOTES ==

General Changes
- Add function :func:`map_key_exists`
```

### CODE REVIEW

**Comment by steveburnett** [comment]
Please add documentation for the new function, following the [Review and Commit guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#documentation).

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestMapKeyExists.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    public void testEmpty()
+    {
+        assertFunction("MAP_KEY_EXISTS(MAP(ARRAY[], ARRAY[]), 5)", BOOLEAN, false);
+    }
+
+    @Test
+    public void testNull()
+    {
+        assertFunction("MAP_KEY_EXISTS(NULL, 1)", BOOLEAN, null);
+    }
```
Comment (kaikalur) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestMapKeyExists.java:55]: Also test for just value being null:

MAP_KEY_EXISTS(MAP(ARRAY[1, null], 1,) -> true
MAP_KEY_EXISTS(MAP(ARRAY[1, null], 2,) -> false

**Review by kaikalur — APPROVED** [review]

**Comment by NikhilCollooru** [comment]
LGTM. Please add documentation, tests as suggested by Sreeni and push the changes.

**Review by kaikalur — DISMISSED** [review]

File: presto-docs/src/main/sphinx/functions/map.rst [hunk 2]
```diff
@@ -99,6 +99,12 @@ Map Functions
         SELECT map_subset(MAP(ARRAY[1,2], ARRAY['a','b']), ARRAY[]); -- {}
         SELECT map_subset(MAP(ARRAY[], ARRAY[]), ARRAY[1,2]); -- {}
 
+.. function:: map_key_exists(x(K, V), k) -> boolean
+
+    Returns whether the give key exists in the map. Returns true if key is present in the input map; false otherwise.
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/functions/map.rst:104]:
```suggestion
    Returns whether the given key exists in the map. Returns ``true`` if key is present in the input map, returns ``false`` if not present.
```
Minor formatting and rephrasing suggestion.

**Review by steveburnett — CHANGES_REQUESTED** [review]
Thanks for the documentation! I had minor formatting and rephrasing suggestions. Let me know what you think, please.

**Review by NikhilCollooru — APPROVED** [review]

**Comment by jainavi17** [comment]
@steveburnett Thanks for reviewing. All the comments have been addressed.

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestMapKeyExists.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                true);
+        assertFunction(
+                "MAP_KEY_EXISTS(MAP(ARRAY[-1, -2, -3], ARRAY[4, 5, 6]), 2)",
+                BOOLEAN,
+                false);
+        assertFunction(
+                "MAP_KEY_EXISTS(MAP(ARRAY['ab', 'bc', 'cd'], ARRAY['x', 'y', 'z']), 'abc')",
+                BOOLEAN,
+                false);
+        assertFunction(
```
Comment (tdcmeehan) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestMapKeyExists.java:39]: Can you add a test case where the key is a NaN?

Comment (kaikalur) [reply presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestMapKeyExists.java:39]: > Can you add a test case where the key is a NaN?

That's probably undefined behavior? Using float/double as map keys is just undefined due to precision issue. So  that won't add any benefit apart from creating a flaky test.

Comment (tdcmeehan) [reply presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestMapKeyExists.java:39]: While behavior over NaN is broadly undefined, people develop dependencies on undefined behavior, and the test cases actually would be deterministic, so I don't know why the test would be flaky.  BTW, there's a test on reals (without NaN) below.

**Review by steveburnett — APPROVED** [review]
LGTM! (docs)

Pull updated branch, local build of docs, looks good. Thanks!

**Comment by jainavi17** [comment]
@tdcmeehan if you could take a look at the PR please. TIA.

**Review by rschlussel — APPROVED** [review]

**Comment by jainavi17** [comment]
@kaikalur, @NikhilCollooru  can we merge this PR please.
